app/lib/features/events/pages/event_details_page.dart

```diff
@@ -0,0 +1,412 @@
+        actions.add(
+          PopupMenuItem(
+            onTap: () => context.pushNamed(
+              Routes.editCalendarEvent.name,
+              pathParameters: {'calendarId': widget.calendarId},
+            ),
+            child: const Row(
+              children: <Widget>[
+                Icon(Atlas.pencil_edit_thin),
+                SizedBox(width: 10),
+                Text('Edit Event'),
+              ],
+            ),
+          ),
+        );
+      }
+
+      //Delete Event Action
+      if (member.canString('CanRedactOwn') &&
+          member.userId().toString() == event.sender().toString()) {
+        final roomId = event.roomIdStr();
+        actions.addAll([
+          PopupMenuItem(
+            onTap: () => showAdaptiveDialog(
+              context: context,
+              builder: (context) => RedactContentWidget(
+                title: 'Remove this post',
+                eventId: event.eventId().toString(),
+                onSuccess: () {
+                  ref.invalidate(calendarEventProvider);
+                  if (context.mounted) {
+                    context.goNamed(
+                      Routes.spaceEvents.name,
+                      pathParameters: {'spaceId': roomId},
+                    );
+                  }
+                },
+                senderId: event.sender().toString(),
+                roomId: roomId,
+                isSpace: true,
+              ),
+            ),
+            child: Row(
+              children: <Widget>[
+                Icon(
+                  Atlas.trash_can_thin,
+                  color: Theme.of(context).colorScheme.error,
+                ),
+                const SizedBox(width: 10),
+                const Text('Remove Event'),
+              ],
+            ),
+          ),
+        ]);
+      }
+    }
+
+    //Report Event Action
+    else {
+      actions.add(
+        PopupMenuItem(
+          onTap: () => showAdaptiveDialog(
+            context: context,
+            builder: (ctx) => ReportContentWidget(
+              title: 'Report this Event',
+              description:
+                  'Report this content to your homeserver administrator. Please note that your administrator won\'t be able to read or view files in encrypted spaces.',
+              eventId: widget.calendarId,
+              roomId: event.roomIdStr(),
+              senderId: event.sender().toString(),
+              isSpace: true,
+            ),
+          ),
+          child: Row(
+            children: <Widget>[
+              Icon(
+                Atlas.warning_thin,
+                color: Theme.of(context).colorScheme.error,
+              ),
+              const SizedBox(width: 10),
+              const Text('Report Event'),
+            ],
+          ),
+        ),
+      );
+    }
+
+    if (actions.isEmpty) {
+      return const SizedBox.shrink();
+    }
+
+    return PopupMenuButton(
+      itemBuilder: (ctx) => actions,
+    );
```

Comment:
as this only happens later, why not push all that menu generation off to when the builder is called?

Reply:
Sorry but not getting you. Can you please more details here?

Reply:
not that important ...